File: GameEngine_PC/GameEngineOpenGL3_2/src/main/java/gameEngine/modelGenerators/WorldAudioGenerator.java
package gameEngine.modelGenerators;

import java.util.HashMap;

import gameEngine.audioEngine.TAudioEnum;
import gameEngine.models.AudioBuffer;
import gameEngine.renderEngine.Loader;

/**
 * Generates the sounds that is going to be used by the audio engine of the game
 */
public class WorldAudioGenerator {

	/**
	 * Folder where the resources are
	 */
	private final static String AUDIO_FOLDER = "audio/";
	
	/**
	 * Creates one dictionary with buffers to the audio engine
	 */
	public static HashMap<TAudioEnum, AudioBuffer> getBuffers(Loader loader) {
		TAudioEnum[] audioTypes = TAudioEnum.values();
		HashMap<TAudioEnum, AudioBuffer> audioLibrary = new HashMap<TAudioEnum, AudioBuffer>();
		
		for(TAudioEnum tAudio : audioTypes) {
			String fileName = tAudio.getValue();
			AudioBuffer audioBuffer = loader.loadSound(AUDIO_FOLDER + fileName);
			if(audioBuffer != null) {
				audioLibrary.put(tAudio, audioBuffer);
			}
		}
		
		return audioLibrary;
	}

}

File: GameEngine_IOS/GameEngine/ModelGenerators/WorldAudioGenerator.swift
import Foundation


/// Generates the sounds that is going to be used by the audio engine of the game
open class WorldAudioGenerator {
    
    /// Creates one dictionary with buffers to the audio engine
    ///
    /// - Parameter loader: The loader of the sounds
    /// - Returns: An dictionary with elements that were possible to read
    open static func getBuffers(_ loader : AudioLoader) -> [TAudioEnum : AudioBuffer] {
        
        var audioLibrary : [TAudioEnum : AudioBuffer] = [TAudioEnum :  AudioBuffer]();
        for tAudio in TAudioEnum.allValues {
            let fileName : String = tAudio.rawValue
            let audioBuffer : AudioBuffer! = loader.load(fileName);
            if(audioBuffer != nil) {
                audioLibrary[tAudio] = audioBuffer
            }
        }
        
        return audioLibrary;
    }
    
}
